### ISSUE TITLE
ftp-sync: uploading... stuck

### ISSUE THREAD

**Issue by rainwilds** [lukasz-wronski/vscode-ftp-sync#375]
I often get an issue in Visual Studio Code where I save the current file and get a notice saying "ftp-sync: uploading..." but nothing happens. The only way to fix the issue is the restart VSC and force an upload.

Any idea why this happens?

Thanks,
David

**Comment by rainwilds** [comment]
![image](https://github.com/lukasz-wronski/vscode-ftp-sync/assets/39784786/11e9c4da-541b-4fe4-864d-e2e2a2f8bdbe)

Just noticed I get this error message.

**Comment by HankLloydRight** [comment]
Yup.. happens to me about once a month, just have to restart VCC. 

It usually takes me a few minutes to figure out why all my changes aren't being made to my uploaded code... then it hits me "yeah, time to restart VCC"

**Comment by rainwilds** [comment]
> Yup.. happens to me about once a month, just have to restart VCC.
> 
> It usually takes me a few minutes to figure out why all my changes aren't being made to my uploaded code... then it hits me "yeah, time to restart VCC"

Thanks for the feedback. Nice to know it's not just me.

I contacted my web server guys and they said I'm maxing out the number of FTP connections (8 in total) allowed. They said most likely the plugin is not closing connections after saving and re-opening a new one every time.

**Comment by HankLloydRight** [comment]
I don’t think it’s leaving connections open… I don’t have any connection limits on my servers and it still locks up. Also you should be using SFTP and not just FTP - which is phenomenally insecure. 

If you are using SFTP than the “FTP limit” shouldn’t affect you.  I doubt the webserver limits SSH connections.

**Comment by HankLloydRight** [comment]
Also you can easily test this 8 connection limit—the plugin should lockup after 8 consecutive saves. I’d guess it’s much higher than that. My guess is a memory leak in the plugin itself -and not related to any type of connection limit(s)

**Comment by rainwilds** [comment]
I tried setting up SSH SFTP. Got it working with a private key in FileZilla but can't get it going with this plugin...
